src/prompts.ts

```diff
@@ -1,15 +1,39 @@
+import type OverridePrompt from "./interfaces/override_prompt";
+
 /**
  * Prompt an AI to convert a given input from one language to another
  * @param inputLanguage - The language of the input
  * @param outputLanguage - The language of the output
  * @param input - The input to be translated
+ * @param overridePrompt - An optional custom prompt
  * @returns A prompt for the AI to translate the input
  */
 export function generationPrompt(
     inputLanguage: string,
     outputLanguage: string,
     input: string,
+    overridePrompt?: OverridePrompt,
 ): string {
+    const customPrompt = overridePrompt?.generationPrompt;
+    const requiredArguments = ["inputLanguage", "outputLanguage", "input"];
+    if (customPrompt) {
+        for (const arg of requiredArguments) {
+            if (!customPrompt.includes(`\${${arg}}`)) {
+                throw new Error(`Missing required argument: \${${arg}}`);
+            }
+        }
+
+        const argumentToValue: { [key: string]: string } = {
+            input,
+            inputLanguage,
+            outputLanguage,
+        };
+
+        return customPrompt.replace(/\$\{([^}]+)\}/g, (match, key) =>
```

Comment:
nit - can we add a comment to show the expected input and output?

Reply:
Sure, I can add a comment explaining this one. It's a templated string in JS like `${this}`